crates/iota-data-ingestion-core/src/worker_pool.rs

```diff
@@ -195,23 +212,48 @@ impl<W: Worker + 'static> WorkerPool<W> {
                             info!("received checkpoint for processing {} for workflow {}", sequence_number, task_name);
                             let start_time = Instant::now();
                             let backoff = backoff::ExponentialBackoff::default();
-                            backoff::future::retry(backoff, || async {
-                                worker
-                                    .clone()
+                            let result = backoff::future::retry(backoff, || async {
```

Comment:
`status` seems more appropriate as we `expect` at the end.
```suggestion
                            let status = backoff::future::retry(backoff, || async {
```